packages/autorest.go/src/generator/operations.ts

```diff
@@ -414,31 +416,35 @@ function generateOperation(op: Operation, imports: ImportManager): string {
     text += '}\n\n';
     return text;
   }
-  const zeroResp = getZeroReturnValue(op, 'op');
+  if (!isLROOperation(op) && injectSpans) {
+    text += `\tctx, endSpan := runtime.StartSpan(ctx, "${clientName}.${opName}", client.internal.Tracer(), nil)\n`;
+    text += '\tdefer func() { endSpan(err) }()\n';
+  }
   text += `\treq, err := client.${info.protocolNaming.requestMethod}(${reqParams})\n`;
   text += `\tif err != nil {\n`;
-  text += `\t\treturn ${zeroResp}, err\n`;
+  text += `\t\treturn\n`;
```

Comment:
I've seen enough concern about bare returns to think we should probably avoid them even in generated code, to forestall questions/complaints about them.

Reply:
Generally speaking, I agree with you (I'm not a fan of named return values).  We could retool this to emit the calls to `endSpan(err)` on all of the error paths, it might get a bit ugly though (although perhaps not as ugly as it is now).  Using named return values does simplify this though.  I will take a closer look.

Reply:
Couldn't we simply `return err` in these cases?

Reply:
Unfortunately no, as `endSpan(err)` terminates the span (success or failure).

Reply:
Here are the ways we could do it.  Not that bad really.  I think I prefer option 2 as we never have to remember to inject calls to `endSpan(err)` if/when we add any additional returns (unlikely but possible).  However, we have to be careful that we don't shadow `err`.  I don't have a strong preference though so could be convinced otherwise.
```go
func (client *AvailabilitySetsClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, availabilitySetName string, parameters AvailabilitySet, options *AvailabilitySetsClientCreateOrUpdateOptions) (AvailabilitySetsClientCreateOrUpdateResponse, error) {
	ctx, endSpan := runtime.StartSpan(ctx, "AvailabilitySetsClient.CreateOrUpdate", client.internal.Tracer(), nil)
	req, err := client.createOrUpdateCreateRequest(ctx, resourceGroupName, availabilitySetName, parameters, options)
	if err != nil {
		endSpan(err)
		return AvailabilitySetsClientCreateOrUpdateResponse{}, err
	}
	resp, err := client.internal.Pipeline().Do(req)
	if err != nil {
		endSpan(err)
		return AvailabilitySetsClientCreateOrUpdateResponse{}, err
	}
	if !runtime.HasStatusCode(resp, http.StatusOK) {
		err := runtime.NewResponseError(resp)
		endSpan(err)
		return AvailabilitySetsClientCreateOrUpdateResponse{}, err
	}
	res, err := client.createOrUpdateHandleResponse(resp)
	endSpan(err)
	return res, err
}

func (client *AvailabilitySetsClient) CreateOrUpdate2(ctx context.Context, resourceGroupName string, availabilitySetName string, parameters AvailabilitySet, options *AvailabilitySetsClientCreateOrUpdateOptions) (AvailabilitySetsClientCreateOrUpdateResponse, error) {
	var err error
	ctx, endSpan := runtime.StartSpan(ctx, "AvailabilitySetsClient.CreateOrUpdate", client.internal.Tracer(), nil)
	defer func() { endSpan(err) }()
	var req *policy.Request
	req, err = client.createOrUpdateCreateRequest(ctx, resourceGroupName, availabilitySetName, parameters, options)
	if err != nil {
		return AvailabilitySetsClientCreateOrUpdateResponse{}, err
	}
	var resp *http.Response
	resp, err = client.internal.Pipeline().Do(req)
	if err != nil {
		return AvailabilitySetsClientCreateOrUpdateResponse{}, err
	}
	if !runtime.HasStatusCode(resp, http.StatusOK) {
		err = runtime.NewResponseError(resp)
		return AvailabilitySetsClientCreateOrUpdateResponse{}, err
	}
	var res AvailabilitySetsClientCreateOrUpdateResponse
	res, err = client.createOrUpdateHandleResponse(resp)
	return res, err
}
```

Reply:
I overthought option 2 a bit.  We don't need to declare all vars up front, just `err`.  So I'll go with that.